lambdaFiles/mainLambda.js

```diff
@@ -19,28 +29,62 @@ function sendMessage(text, channel) {
   https.get(options);
 }
 
+function putObjectInS3(text) {
+  var s3 = new AWS.S3();
+  var params = {
+    Bucket: config.S3_BUCKET_NAME,
+    Key: "boba-buddies/text.json",
+    Body: JSON.stringify({ text }),
+    ContentType: "application/json; charset=utf-8",
+  };
+  s3.putObject(params, (err, data) => {
+    if (err) console.log(err);
+    else console.log(data);
+  });
+}
+
 /**
  * resends the direct message being sent to it
  */
 function resendText(event, callback) {
   // test ensures that message isnt from a bot and is a direct message
+
+  console.log(event.bot_id, !event.bot_id);
   if (!event.bot_id && event.channel_type === "im") {
     const { text, channel } = event;
     sendMessage(text, channel);
+    putObjectInS3(text);
+    callback(undefined, responseSuccess);
+  } else {
+    console.log("don't respond to self");
+    callback(undefined, responseSuccess);
   }
-  callback(null);
+}
+
+function handleInteractions(payload, callback) {
```

Comment:
Cracked; should we put the data into s3 from here as well?

Reply:
On top of that, we'd probably need a if statement when the yes or no button is pressed. Lol if pressed no, then don't do anything?